projects/step-core/src/lib/modules/table/components/table/table.component.ts

```diff
@@ -113,6 +113,8 @@ export class TableComponent<T> implements AfterViewInit, OnChanges, OnDestroy, T
 
   searchColumns: SearchColumn[] = [];
 
+  pageSizeOptions: Array<number>;
```

Comment:
this is something I believe we should actually discuss because it raises a valid point - as I see, you don't have ? or ! after ```pageSizeOptions```, and the reason you are not getting a compilation error is because it is assigned in the constructor; I think we should consider whether we should start making more use of our constructors for code that does not directly relate to the component lifecycle

Reply:
if we talk about components, code that can be put in constructors can also be seamlessly put in ```ngOnInit``` (at least most of times); if we talk about services, they don't have lifecycle hooks (except for ```ngOnDestroy``` if they are provided at the component level), so the code goes in the constructor; there is one more alternative - code that would go into a service's constructor could be extracted into a method, calling which would act as a constructor, but the downside is that in order to assign ```readonly``` properties in such a function we would have to use the ```(this as { foo: boolean }).foo = true``` approach, which we could save ourselves from if we just keep those initializations in the constructor (as it is totally fine to initialize ```readonly``` properties in the constructor)

Reply:
I will create a card on Trello for these aforementioned points and I will refer to this PR's discussion